src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/SqlCommand/SqlCommandCancelTest.cs

```diff
@@ -558,5 +573,50 @@ private static async Task AsyncCancelDoesNotWait(string connStr)
                 Assert.InRange((ended - started).TotalSeconds, cancelSeconds, delaySeconds - 1);
             }
         }
+
+        private static void AsyncCancelDoesNotWait2(string connStr)
+        {
+            const int delaySeconds = 30;
+            const int cancelSeconds = 1;
+
+            var cancellationTokenSource = new CancellationTokenSource();
+            DateTime started = DateTime.UtcNow;
+            DateTime ended = DateTime.UtcNow;
+            Exception exception = null;
+
+            Task executing = ExecuteWaitForAsync(cancellationTokenSource.Token, connStr, delaySeconds);
+
+            cancellationTokenSource.CancelAfter(TimeSpan.FromSeconds(cancelSeconds+1));
+
+            try
+            {
+                executing.Wait();
+            }
+            catch (Exception ex)
+            {
+                exception = ex;
+            }
+            ended = DateTime.UtcNow;
+
+            Assert.NotNull(exception);
+            Assert.InRange((ended - started).TotalSeconds, cancelSeconds, delaySeconds - 1);
+        }
+
+        private static async Task ExecuteWaitForAsync(CancellationToken cancellationToken, string connectionString, int delaySeconds)
+        {
+            using (var connection = new SqlConnection(connectionString))
+            {
+                await connection.OpenAsync().ConfigureAwait(false);
+                using (var command = new SqlCommand(@"
+WHILE 1 = 1
+BEGIN
+    DECLARE @x INT = 1
+END", connection))
+                {
+                    command.CommandTimeout = delaySeconds + 10;
+                    await command.ExecuteNonQueryAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
To maximize the code coverage I'd suggest running all the `command.ExecuteXXXAsync` functions.